src/conic_form.jl

```diff
@@ -0,0 +1,270 @@
+mutable struct ConeData
+    f::Int # number of linear equality constraints
+    l::Int # length of LP cone
+    q::Int # length of SOC cone
+    qa::Vector{Int} # array of second-order cone constraints
+    s::Int # length of SD cone
+    sa::Vector{Int} # array of semi-definite constraints
+    ep::Int # number of primal exponential cone triples
+    ed::Int # number of dual exponential cone triples
+    p::Vector{Float64} # array of power cone params
+    nrows::Dict{Int, Int} # The number of rows of each vector sets, this is used by `constrrows` to recover the number of rows used by a constraint when getting `ConstraintPrimal` or `ConstraintDual`
+    function ConeData()
+        new(0, 0, 0, Int[], 0, Int[], 0, 0, Float64[], Dict{Int, Int}())
+    end
+end
+
+mutable struct ModelData
+    m::Int # Number of rows/constraints
+    n::Int # Number of cols/variables
+    I::Vector{Int} # List of rows
+    J::Vector{Int} # List of cols
+    V::Vector{Float64} # List of coefficients
+    b::Vector{Float64} # constants
+    objective_constant::Float64 # The objective is min c'x + objective_constant
+    c::Vector{Float64}
+end
+
+mutable struct ConicData
```

Comment:
At first sight, the exact difference between `ConeData` and `ConicData` looks unclear to me

Reply:
makes sense. removed merged `ConeData` fields to conicdata